ResearchKit/Common/ORKHealthAnswerFormat.m

```diff
@@ -154,26 +164,52 @@ - (ORKAnswerFormat *)impliedAnswerFormat {
             impliedAnswerFormat = format;
             
         } else if ([identifier isEqualToString:HKCharacteristicTypeIdentifierDateOfBirth]) {
-            NSCalendar *calendar = [NSCalendar currentCalendar];
-            NSDate *now = [NSDate date];
-            NSDate *thirtyFiveYearsAgo = [calendar dateByAddingUnit:NSCalendarUnitYear value:-35 toDate:now options:0];
-            NSDate *minDate = [calendar dateByAddingUnit:NSCalendarUnitYear value:-150 toDate:now options:0];
-            NSDate *maxDate = [calendar dateByAddingUnit:NSCalendarUnitDay value:1 toDate:now options:0];
-            
-            ORKDateAnswerFormat *format = [ORKDateAnswerFormat dateAnswerFormatWithDefaultDate:thirtyFiveYearsAgo
-                                                                           minimumDate:minDate
-                                                                           maximumDate:maxDate
-                                                                              calendar:nil];
+            ORKDateAnswerFormat *format = [ORKDateAnswerFormat dateAnswerFormatWithDefaultDate:_defaultDate
+                                                                           minimumDate:_minimumDate
+                                                                           maximumDate:_maximumDate
+                                                                              calendar:_calendar];
             impliedAnswerFormat = format;
         }
     }
     return impliedAnswerFormat;
 }
 
+- (void)setDefaultDate:(NSDate *)defaultDate {
+    if (!defaultDate) {
+        defaultDate = [_calendar dateByAddingUnit:NSCalendarUnitYear value:-35 toDate:[NSDate date] options:0];
```

Comment:
Regarding archiving and unarchiving you're probably right. I'll change the implementation so these properties are `nullable` instead of `null_resettable`.

---

Regarding intervals, what about if you want to target a date which is not around present time (e.g., people between 50 and 60)? Then just an interval is not enough. I modelled the arguments mirroring `ORKDateAnswerFormat`.

---

Regarding default value, you are right, I can make it so the default values are calculated on reading instead on writting.